src/pages/PeoplePage.tsx

```diff
@@ -0,0 +1,94 @@
+import { useEffect, useState } from 'react';
+import { Loader } from '../components/Loader';
+import { PeopleTable } from '../components/PeopleTable/PeopleTable';
+import { Person } from '../types';
+import { getPeople } from '../api';
+import { PeopleFilters } from '../components/PeopleFilters';
+import { useSearchParams } from 'react-router-dom';
+import { getPeopleWithParents } from '../utils/getPeopleWithParents';
+import { getPreparedPeople } from '../utils/getPreparedPeople';
+import { SearchParams } from '../types/Enums';
+
+export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [hasError, setHasError] = useState(false);
+  const [searchParams] = useSearchParams();
+  const activeSex = searchParams.get(SearchParams.Sex);
+  const filterQuery = searchParams.get(SearchParams.Query);
+  const activeSort = searchParams.get(SearchParams.Sort);
+  const isDesc = searchParams.get(SearchParams.Order);
+  const activeCenturies = searchParams.getAll(SearchParams.Centuries);
+
+  const loadPeople = () => {
+    getPeople()
+      .then(peopleFromServer => {
+        const preparedPeople = getPeopleWithParents(peopleFromServer);
+
+        setPeople(preparedPeople);
+      })
+      .catch(() => setHasError(true))
+      .finally(() => setIsLoading(false));
+  };
+
+  const filteredPeople = getPreparedPeople(
+    people,
+    activeSex,
+    filterQuery,
+    activeCenturies,
+    activeSort,
+    isDesc,
+  );
+
+  useEffect(() => {
+    setIsLoading(true);
+    setHasError(false);
+    loadPeople();
+  }, []);
```

Comment:
are you afraid to move it to the middle?))
```suggestion
  const filteredPeople = getPreparedPeople(
    people,
    activeSex,
    filterQuery,
    activeCenturies,
    activeSort,
    isDesc,
  );

  useEffect(() => {
    setIsLoading(true);
    setHasError(false);
    
    getPeople()
      .then(peopleFromServer => {
        const preparedPeople = getPeopleWithParents(peopleFromServer);

        setPeople(preparedPeople);
      })
      .catch(() => setHasError(true))
      .finally(() => setIsLoading(false));
  }, []);
```